hathor/utils/dict.py

```diff
@@ -0,0 +1,29 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+def deep_merge(first: dict, second: dict) -> None:
+    """
+    Recursively merges two dicts, altering the first one in place.
```

Comment:
Add a note that this method will fail if there's a circular reference in `first` and `second`. More specifically, it will be an infinite recursion until maximum recursion depth is exceeded.

For instance:

```python
>>> a = {}
>>> b = {}
>>> a['x'] = b
>>> b['x'] = a
>>> deep_merge(a, b)
```

I don't think we have to handle this case and a note is enough for our use.

Reply:
Makes sense, added in https://github.com/HathorNetwork/hathor-core/pull/618/commits/32b1a3444a4c4c916f89713fed7fe8886f358e69

Reply:
Great! Thanks!